src/service/resource.js

```diff
@@ -932,30 +939,45 @@ export class Resource {
     dev().fine(TAG, 'start layout:', this.debugid, 'count:', this.layoutCount_);
     this.layoutCount_++;
     this.state_ = ResourceState.LAYOUT_SCHEDULED;
+    this.abortController_ = new AbortController();
+    const abortSignal = this.abortController_.signal;
 
     const promise = new Promise((resolve, reject) => {
       Services.vsyncFor(this.hostWin).mutate(() => {
+        if (abortSignal.aborted) {
+          // The chained layoutCount_ will log the expected race error.
+          return;
+        }
         try {
-          resolve(this.element.layoutCallback());
+          resolve(this.element.layoutCallback(abortSignal));
         } catch (e) {
           reject(e);
         }
       });
-    });
-
-    this.layoutPromise_ = promise.then(
-      () => this.layoutComplete_(true),
-      (reason) => this.layoutComplete_(false, reason)
+    }).then(
+      () => this.layoutComplete_(true, abortSignal),
+      (reason) => this.layoutComplete_(false, abortSignal, reason)
     );
-    return this.layoutPromise_;
+
+    return (this.layoutPromise_ = promise);
   }
 
   /**
    * @param {boolean} success
+   * @param {!AbortSignal} abortSignal
    * @param {*=} opt_reason
    * @return {!Promise|undefined}
    */
-  layoutComplete_(success, opt_reason) {
+  layoutComplete_(success, abortSignal, opt_reason) {
+    if (abortSignal.aborted) {
+      // We hit a race condition, where `layoutCallback` -> `unlayoutCallback`
```

Comment:
Do we need to unschedule the element? Otherwise it will never get rescheduled for loading again, right?